Job control signal forwarding without setsid test is flaky

https://circleci.com/gh/chriskuehl/dumb-init/95 failed due to test flakes:

* httpredir redirected us to a bad mirror (not much can be done about this, we are very familiar with this problem...)
* Our test for job control signal forwarding with setsid disabled is flaky:

```
=================================== FAILURES ===================================
____________________________ test_prints_signals[1] ____________________________

both_debug_modes = None, setsid_disabled = None

    def test_prints_signals(both_debug_modes, setsid_disabled):
        """In non-setsid mode, dumb-init should forward the signals SIGTSTP,
        SIGTTOU, and SIGTTIN, and then suspend itself.
        """
        proc = Popen(
            ('dumb-init', sys.executable, '-m', 'tests.lib.print_signals'),
            stdout=PIPE,
        )
    
        assert re.match(b'^ready \(pid: (?:[0-9]+)\)\n$', proc.stdout.readline())
    
        for signum in SUSPEND_SIGNALS:
            assert process_state(proc.pid) in ['running', 'sleeping']
            proc.send_signal(signum)
            assert proc.stdout.readline() == '{0}\n'.format(signum).encode('ascii')
>           assert process_state(proc.pid) == 'stopped'
E           assert 'running' == 'stopped'
E             - running
E             + stopped

tests/shell_background_test.py:73: AssertionError
----------------------------- Captured stderr call -----------------------------
[dumb-init] Running in debug mode.
[dumb-init] Not running in setsid mode.
[dumb-init] Child spawned with PID 5001.
[dumb-init] Received signal 20.
[dumb-init] Not running in setsid mode, so forwarding the original signal (20).
[dumb-init] Forwarded signal 20 to children.
[dumb-init] Suspending self due to TTY signal.
===================== 1 failed, 94 passed in 2.29 seconds ======================
```

The equivalent test for setsid mode (behavior is a bit different due to [kernel limitations](https://github.com/Yelp/dumb-init/blob/v1.0.1/dumb-init.c#L65-L77)) sleeps a bit. It's pretty obvious why there is a race here. We can probably avoid sleeping altogether using `wait` (to check for the process suspending/resuming due to signals).

--- a/tests/proxies_signals_test.py
+++ b/tests/proxies_signals_test.py
@@ -9,7 +9,7 @@ from tests.lib.testing import NORMAL_SIGNALS
 from tests.lib.testing import pid_tree
 
 
-def test_prints_signals(both_debug_modes, both_setsid_modes):
+def test_proxies_signals(both_debug_modes, both_setsid_modes):
     """Ensure dumb-init proxies regular signals to its child."""
     proc = Popen(
         ('dumb-init', sys.executable, '-m', 'tests.lib.print_signals'),

--- a/tests/shell_background_test.py
+++ b/tests/shell_background_test.py
@@ -55,7 +55,7 @@ def test_shell_background_support_setsid(both_debug_modes, setsid_enabled):
         os.kill(pid, SIGKILL)
 
 
-def test_prints_signals(both_debug_modes, setsid_disabled):
+def test_shell_background_support_without_setsid(both_debug_modes, setsid_disabled):
     """In non-setsid mode, dumb-init should forward the signals SIGTSTP,
     SIGTTOU, and SIGTTIN, and then suspend itself.
     """
@@ -70,6 +70,7 @@ def test_prints_signals(both_debug_modes, setsid_disabled):
         assert process_state(proc.pid) in ['running', 'sleeping']
         proc.send_signal(signum)
         assert proc.stdout.readline() == '{0}\n'.format(signum).encode('ascii')
+        os.waitpid(proc.pid, os.WUNTRACED)
         assert process_state(proc.pid) == 'stopped'
 
         proc.send_signal(SIGCONT)

